README.md

```diff
@@ -152,14 +152,47 @@ public class Startup
 {
   public void ConfigureServices(IServiceCollection services)
   {
-      services.AddFeatureManagement();
+      services.AddFeatureManagement()
+              .AddFeatureFilter<MyCriteriaFilter>();
   }
 }
 ```
 
-This tells the feature manager to use the "FeatureManagement" section from the configuration for feature flag settings.
+This tells the feature manager to use the "FeatureManagement" section from the configuration for feature flag settings. It also registers a customized feature filter `MyCriteriaFilter`. Please refer to the `Implementing a Feature Filter` section below for more details.
 
-**Advanced:** The feature manager looks for feature definitions in a configuration section named "FeatureManagement". If the "FeatureManagement" section does not exist, the configuration will be considered empty.
+You can also specify that feature management configuration should be retrieved from a different configuration section by calling configuration.GetSection and passing in the name of the desired section. The following example tells the feature manager to read from a different section called "MyFeatureFlags" instead:
+``` C#
+…
+services.AddFeatureManagement(configuration.GetSection("MyFeatureFlags"));
+…
+```
+
+**Advanced:** By default, the feature manager retrieves feature flag configuration from the "FeatureManagement" section of the .NET Core configuration data. If the "FeatureManagement" section does not exist, the configuration will be considered empty. When a specific configuration section is provided to `AddFeatureManagement()`, the feature manager will fall back to the root of that configuration section to look for feature definitions, if there is no "FeatureManagement" section found. 
+
+### Scoped Feature Management Services
+There are scenarios that feature filters are not necessarily be singleton ([#15](https://github.com/microsoft/FeatureManagement-Dotnet/issues/15), [#258](https://github.com/microsoft/FeatureManagement-Dotnet/issues/258)). For example, users may want to use feature filters which consume scoped services for context information. In this case, the feature management should be registered as scoped by calling `AddScopedFeatureManagement`.
+
+This example demonstrates how to register feature manager and feature filters as scoped services to allow the feature filter `MyCriteriaFilter` to consume the scoped context provider service `MyContextProvider`.
+``` C#
+public class MyCriteriaFilter : IFeatureFilter
+{
+    private readonly IContextProvider _contextProvider;
+
+    public MyCriteriaFilter(IContextProvider contextProvider)
+    {
+        _contextProvider = contextProvider;
+    }
+}
+```
+
+``` C#
+…
+services.AddScopedFeatureManagement()
+        .AddFeatureFilter<MyCriteriaFilter>();

```

Comment:
I'd leave out the filter. This is kind of an "all-else-follows" section. Just mention that feature manager, as well as all registered filters are added as scoped.

Reply:
Here I want to present an example which demonstrates how to register feature manager and feature filters as scoped services to allow the feature filter to consume the scoped context provider service

Are you suggesting that I should remove this example?